deployments/helm/nvidia-dra-driver-gpu/templates/kubeletplugin.yaml

```diff
@@ -49,6 +49,70 @@ spec:
       serviceAccountName: {{ include "nvidia-dra-driver-gpu.serviceAccountName" . }}
       securityContext:
         {{- toYaml .Values.kubeletPlugin.podSecurityContext | nindent 8 }}
+      initContainers:
+      - name: validate-depenencies
+        image: {{ include "nvidia-dra-driver-gpu.fullimage" . }}
+        command: ["bash", "-c"]
+        args:
+        - |-
+          # Ensure that
+          #
+          # 1) GPU driver binaries are in the NVIDIA_DRIVER_ROOT directory.
+          # 2) the GPU driver is set up (nvidia-smi reports health).
+          #
+          # Main intent it to help troubleshoot a setup that does not have the
+          # GPU driver set up properly before installing this DRA driver. In
+          # that case, this container log is meant to yield an actionable error
+          # message. Another goal is to crash-loop (retry periodically) and
+          # actively wait for the GPU driver to be set up properly (this init
+          # container is meant to be a long-running init container that only
+          # exits upon success). That allows for auto-healing the DRA driver
+          # instalaltion right after the GPU driver setup has been fixed.
+          #
+          # For checking GPU driver health: for now, rely on nvidia-smi's exit
+          # code. Rely on code 0 meaning that the driver is properly set up. For
+          # example, code 9 means that the GPU driver is not loaded; see section
+          # 'RETURN VALUE' in the man page.
+          #
+          echo "NVIDIA_DRIVER_ROOT (path on host): ${NVIDIA_DRIVER_ROOT}"
+          echo "command -v nvidia-smi (path in container): $(command -v nvidia-smi)"
+          while :
+          do
+          if ! command -v nvidia-smi &>/dev/null
+          then
```

Comment:
I think doing this in bash is fine, but it may be worthwhile considering writing the logic in a script that we embed in the contaienr and just call out to here, instead of inlining everything.